### PR TITLE
Dmp 2057 arm event date calculation

### PR DESCRIPTION
**Pull request by karen-hedges**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DMP-2057

### Change description ###

Created automated task to calculate event date and send to ARM

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/arm/component/impl/ArmRetentionEventDateCalculatorImpl.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    private UserAccountEntity userAccount;
+
+    @Transactional
+    public void calculateRetentionEventDate(Integer externalObjectDirectoryId) {
+        userAccount = userIdentity.getUserAccount();
+        try {
+            ExternalObjectDirectoryEntity externalObjectDirectory = externalObjectDirectoryRepository.findById(externalObjectDirectoryId).orElseThrow();
+            OffsetDateTime retentionDate = getDocumentRetentionDate(externalObjectDirectory);
+            if (nonNull(retentionDate)) {
+                OffsetDateTime armRetentionDate = retentionDate.minusDays(100);
```
Comment (mestebanez) [inline src/main/java/uk/gov/hmcts/darts/arm/component/impl/ArmRetentionEventDateCalculatorImpl.java:39]: Should this 100 be in configuration?

File: src/main/java/uk/gov/hmcts/darts/arm/component/ArmRetentionEventDateCalculator.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package uk.gov.hmcts.darts.arm.component;
+
+public interface ArmRetentionEventDateCalculator {
+
+    void calculateRetentionEventDate(Integer externalObjectDirectoryId);
```
Comment (mestebanez) [inline src/main/java/uk/gov/hmcts/darts/arm/component/ArmRetentionEventDateCalculator.java:5]: Could return a boolean to signify a failure state. Be nice if clients could react to a failure by putting out a log message

Comment (karen-hedges) [reply src/main/java/uk/gov/hmcts/darts/arm/component/ArmRetentionEventDateCalculator.java]: There are logs for any failures. Not sure its relevant for the automated task.

File: src/main/java/uk/gov/hmcts/darts/arm/component/impl/ArmRetentionEventDateCalculatorImpl.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    private UserAccountEntity userAccount;
+
+    @Transactional
+    public boolean calculateRetentionEventDate(Integer externalObjectDirectoryId) {
+        userAccount = userIdentity.getUserAccount();
+        try {
+            ExternalObjectDirectoryEntity externalObjectDirectory = externalObjectDirectoryRepository.findById(externalObjectDirectoryId).orElseThrow();
+            OffsetDateTime retentionDate = getDocumentRetentionDate(externalObjectDirectory);
+            if (nonNull(retentionDate)) {
+                OffsetDateTime armRetentionDate = retentionDate.minus(armDataManagementConfiguration.getEventDateDurationAdjustment());
```
Comment (hemantasharma1129) [inline src/main/java/uk/gov/hmcts/darts/arm/component/impl/ArmRetentionEventDateCalculatorImpl.java:41]: Should be 100 years. In application yaml I can see it as 100d. Is it days?

**Review by hemantasharma1129 — APPROVED** [review]

**Pull request merged**